astropy/wcs/wcs.py

```diff
@@ -388,8 +396,8 @@ def __init__(self, header=None, fobj=None, key=' ', minerr=0.0,
                         "header must be a string, an astropy.io.fits.Header "
                         "object, or a dict-like object")
 
-            header_string = header_string.strip()
-
```

Comment:
:+1:  Do you want me to make a PR or do you prefer to make it? It's either removing these lines or I have to make numerous changes in stwcs.